osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/infra/jms/MessageMetadata.java

```diff
@@ -81,60 +88,51 @@ public static final MessageMetadata fromMessage(final Message message) throws JM
     return metadata;
   }
 
-  public static MessageMetadata.Builder newMessageMetadataBuilder() {
-    return new Builder();
-  }
-
-  public String getDeviceIdentification() {
-    return this.deviceIdentification;
-  }
-
-  public String getOrganisationIdentification() {
-    return this.organisationIdentification;
-  }
-
-  public String getCorrelationUid() {
-    return this.correlationUid;
-  }
+  public void applyTo(final Message message) throws JMSException {
 
-  public String getDomain() {
-    return this.domain;
-  }
-
-  public String getDomainVersion() {
-    return this.domainVersion;
-  }
-
-  public String getIpAddress() {
-    return this.ipAddress;
-  }
+    message.setJMSCorrelationID(this.correlationUid);
+    message.setJMSType(this.messageType);
+    message.setJMSPriority(this.messagePriority);
 
-  public String getMessageType() {
-    return this.messageType;
-  }
+    if (StringUtils.isNotBlank(this.deviceIdentification)) {
+      message.setStringProperty(Constants.DEVICE_IDENTIFICATION, this.deviceIdentification);
+    }
+    if (StringUtils.isNotBlank(this.organisationIdentification)) {
+      message.setStringProperty(
+          Constants.ORGANISATION_IDENTIFICATION, this.organisationIdentification);
+    }
 
-  public int getMessagePriority() {
-    return this.messagePriority;
-  }
+    if (StringUtils.isNotBlank(this.domain)) {
+      message.setStringProperty(Constants.DOMAIN, this.domain);
+    }
+    if (StringUtils.isNotBlank(this.domainVersion)) {
+      message.setStringProperty(Constants.DOMAIN_VERSION, this.domainVersion);
+    }
 
-  public Long getScheduleTime() {
-    return this.scheduleTime;
-  }
+    if (StringUtils.isNotBlank(this.ipAddress)) {
+      message.setStringProperty(Constants.IP_ADDRESS, this.ipAddress);
+    }
 
-  public boolean isScheduled() {
-    return this.scheduled;
-  }
+    if (this.scheduleTime != null) {
+      message.setLongProperty(Constants.SCHEDULE_TIME, this.scheduleTime);
+    }
+    if (this.maxScheduleTime != null) {
+      message.setLongProperty(Constants.MAX_SCHEDULE_TIME, this.maxScheduleTime);
+    }
+    message.setBooleanProperty(Constants.IS_SCHEDULED, this.scheduled);
 
-  public int getRetryCount() {
-    return this.retryCount;
-  }
+    message.setIntProperty(Constants.RETRY_COUNT, this.retryCount);
+    message.setBooleanProperty(Constants.BYPASS_RETRY, this.bypassRetry);
 
-  public boolean isBypassRetry() {
-    return this.bypassRetry;
+    /*
+     * Not setting the jmsxDeliveryCount as int property named Constants.DELIVERY_COUNT, because
+     * this is not some metadata to be transferred over to new JMS messages. This delivery count is
+     * incremented by the message queue environment when a message is re-delivered to a consumer.
+     */
   }
 
-  public int getJmsxDeliveryCount() {
-    return this.jmsxDeliveryCount;
+  public static MessageMetadata.Builder newBuilder() {
```

Comment:
For Builders, a creator called `aSomething` plays nice with the fluent api that follows. `aMessageMetadata.withIpAddress...`

Reply:
I am not entirely sure I know what you mean.
Is it replacing code like:
```java
final ProtocolResponseMessage protocolResponseMessage =
    ProtocolResponseMessage.newBuilder()
        .result(ResponseMessageResultType.NOT_OK)
        .build();
```
with something similar to:
```java
final Builder aResponseMessage = ProtocolResponseMessage.newBuilder();
final ProtocolResponseMessage protocolResponseMessage =
    aResponseMessage
        .withResult(ResponseMessageResultType.NOT_OK)
        .build();
```
If so, I don't think there is much of a reason to. Purely stylistic different preference.
If we would have some agreed standard, that should be followed, but we don't.
At the moment I tend to the more succinct version of the code without the explicit builder variable, and without the with-prefix in all builder methods.

If this is all based on my not understanding your comment, or you think we need to get more into this, please let me know.

Reply:
I meant renaming `newBuilder()` to `aMessageMetadata()`. Parenthesis make all the difference ;)